src/compiler/types.ts

```diff
@@ -2382,19 +2382,16 @@ namespace ts {
         Union                   = 1 << 19,  // Union (T | U)
         Intersection            = 1 << 20,  // Intersection (T & U)
         Anonymous               = 1 << 21,  // Anonymous
-        Instantiated            = 1 << 22,  // Instantiated anonymous type

```

Comment:
I would preserve this flag unless you really need to free it up. In generic code we create *lots* of instantiated anonymous types and it would be nice not to incur the overhead of another property.